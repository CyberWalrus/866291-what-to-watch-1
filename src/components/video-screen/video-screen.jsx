import React, {Fragment} from "react";
import HiddenIcon from "../hidden-icon/hidden-icon.jsx";
import PropTypes from "prop-types";
import VideoPlayer from "../video-player/video-player.jsx";
import {OptionsVideoFull} from "../../mock/constants.js";
import {connect} from "react-redux";
import {getFilm} from "../../store/data/selectors.js";

const VideoScreen = ({onVideoScreenClose, isPlaying, film}) => {
  return (
    <Fragment>
      <HiddenIcon />
      <div
        className="player"
        style={{
          zIndex: `100`,
          background: `url(${film.srcBgImage}) no-repeat center center fixed`,
          backgroundSize: `cover`
        }}
      >
        <VideoPlayer
          options={OptionsVideoFull}
          videoSrc={film.srcVideo}
          posterSrc={film.srcBgImage}
          isPlaying={isPlaying}
        />

        <button
          type="button"
          className="player__exit"
          onClick={onVideoScreenClose}
        >
          Exit
        </button>
      </div>
    </Fragment>
  );
};
VideoScreen.propTypes = {
  filmId: PropTypes.number.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  onPlayChange: PropTypes.func.isRequired,
  onVideoScreenClose: PropTypes.func.isRequired,
  film: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    bgColor: PropTypes.string.isRequired,
    srcPreviewImage: PropTypes.string.isRequired,
    srcPosterImage: PropTypes.string.isRequired,
    srcBgImage: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    srcVideo: PropTypes.string.isRequired,
    rating: PropTypes.string.isRequired,
    scoresCount: PropTypes.number.isRequired,
    ratingLevel: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starrings: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    runTime: PropTypes.number.isRequired
  })
};
const mapStateToProps = (state, ownProps) =>
  Object.assign({}, ownProps, {
    film: getFilm(state, ownProps.filmId)
  });

export {VideoScreen};

export default connect(mapStateToProps)(VideoScreen);
