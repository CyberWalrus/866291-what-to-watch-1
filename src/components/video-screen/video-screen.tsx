import * as React from "react";
import {PureComponent, Fragment, RefObject, ReactElement} from "react";
import VideoPlayer from "../video-player/video-player";
import {OptionsVideoFull, VideoScreenStyle} from "../../constants";
import {connect} from "react-redux";
import {getFilm} from "../../store/data/selectors";
import {Film} from "../../type/data";
import {StateApp} from "../../type/reducer";

interface PropsInsert {
  filmId: number;
  time: string;
  progressValue: number;
  isPlaying: boolean;
  onChangePlay: () => void;
  onClickClose: () => void;
  onClickFullScreen: () => void;
  onMouseTogglerDown: () => void;
  onSendVideoRef: () => void;
  onSendProgressRef: (value: any) => void;
}
interface PropsState {
  film: Film;
}
type Props = PropsInsert & PropsState;
class VideoScreen extends PureComponent<Props, null> {
  public progressRef: RefObject<HTMLProgressElement>;
  public constructor(props: Props) {
    super(props);

    this.progressRef = React.createRef();
  }
  public componentDidMount(): void {
    if (this.props.onSendProgressRef) {
      this.props.onSendProgressRef(this.progressRef.current);
    }
  }
  public render(): ReactElement {
    const {
      isPlaying,
      film,
      time,
      progressValue,
      onClickClose,
      onSendVideoRef,
      onChangePlay,
      onClickFullScreen,
      onMouseTogglerDown
    } = this.props;
    return (
      <Fragment>
        <div
          className="player"
          style={Object.assign({}, VideoScreenStyle, {
            background: `url(${film.srcBgImage}) no-repeat center center fixed`
          })}
        >
          <VideoPlayer
            options={OptionsVideoFull}
            videoSrc={film.srcVideo}
            posterSrc={film.srcBgImage}
            onSendVideoRef={onSendVideoRef}
            isPlaying={isPlaying}
          />

          <button type="button" className="player__exit" onClick={onClickClose}>
            Exit
          </button>
          <div className="player__controls">
            <div className="player__controls-row">
              <div className="player__time">
                <progress
                  ref={this.progressRef}
                  className="player__progress"
                  value={progressValue}
                  max="100"
                />
                <div
                  className="player__toggler"
                  onMouseDown={onMouseTogglerDown}
                  style={{left: `${progressValue}%`}}
                >
                  Toggler
                </div>
              </div>
              <div className="player__time-value">{time}</div>
            </div>

            <div className="player__controls-row">
              {isPlaying ? (
                <button
                  type="button"
                  className="player__play"
                  onClick={onChangePlay}
                >
                  <svg viewBox="0 0 14 21" width="14" height="21">
                    <use xlinkHref="#pause" />
                  </svg>
                  <span>Pause</span>
                </button>
              ) : (
                <button
                  type="button"
                  className="player__play"
                  onClick={onChangePlay}
                >
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s" />
                  </svg>
                  <span>Play</span>
                </button>
              )}

              <div className="player__name">{film.title}</div>

              <button
                type="button"
                className="player__full-screen"
                onClick={onClickFullScreen}
              >
                <svg viewBox="0 0 27 27" width="27" height="27">
                  <use xlinkHref="#full-screen" />
                </svg>
                <span>Full screen</span>
              </button>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = (state: StateApp, ownProps: Props): Props =>
  Object.assign({}, ownProps, {
    film: getFilm(state, ownProps.filmId)
  });

export {VideoScreen};

export default connect<Props, {}, {}, StateApp>(mapStateToProps)(VideoScreen);
