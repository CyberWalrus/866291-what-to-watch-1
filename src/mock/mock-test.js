const FILMS = [
  {
    bgColor: `#977461`,
    description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
    director: `Martin Scorsese`,
    genre: `Thriller`,
    id: 1,
    isFavorite: false,
    pageUrl: `Shutter Island`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    rating: `8,1`,
    ratingLevel: `Very good`,
    released: 2010,
    runTime: 138,
    scoresCount: 1002557,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
    starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
    title: `Shutter Island`
  },
  {
    bgColor: `#F1E9CE`,
    description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
    director: `Justin Kurzel`,
    genre: `Drama`,
    id: 2,
    isFavorite: false,
    pageUrl: `Macbeth`,
    srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    srcPreviewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    rating: `6,6`,
    ratingLevel: `Good`,
    released: 2015,
    runTime: 113,
    scoresCount: 48798,
    srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Macbeth.jpg`,
    srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
    srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/macbeth.jpg`,
    starrings: [(`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`)],
    title: `Macbeth`
  }
];
const FILM = {
  bgColor: `#977461`,
  description: `In 1954, a U.S. Marshal investigates the disappearance of a murderer, who escaped from a hospital for the criminally insane.`,
  director: `Martin Scorsese`,
  genre: `Thriller`,
  id: 1,
  isFavorite: false,
  pageUrl: `Shutter Island`,
  srcVideo: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  srcPreviewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: `8,1`,
  ratingLevel: `Very good`,
  released: 2010,
  runTime: 138,
  scoresCount: 1002557,
  srcBgImage: `https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg`,
  srcPosterImage: `https://es31-server.appspot.com/wtw/static/film/poster/Shutter_Island.jpg`,
  srcPreviewImage: `https://es31-server.appspot.com/wtw/static/film/preview/shutter-island.jpg`,
  starrings: [(`Leonardo DiCaprio`, `Emily Mortimer`, `Mark Ruffalo`)],
  title: `Shutter Island`
};
const REVIEWS = [
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  },
  {
    comment: `test test test test test test test test test test test test test`,
    date: `2019-05-11`,
    dateHTML: `May 11, 2019`,
    id: 1,
    rating: `8,1`,
    userId: 1,
    userName: `Walrus`
  }
];

const GENRES = [
  `All genres`,
  `Thriller`,
  `Drama`,
  `Adventure`,
  `Comedy`,
  `Crime`,
  `Action`,
  `Fantasy`
];
export {FILMS, FILM, GENRES, REVIEWS};
