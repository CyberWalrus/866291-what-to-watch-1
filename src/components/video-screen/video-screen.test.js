import React from "react";
import renderer from "react-test-renderer";
import {VideoScreen} from "./video-screen.jsx";
import {FILM} from "../../mock/mock-test.js";

it(`VideoScreen correctly renders after relaunch`, () => {
  const handleClick = jest.fn();
  const tree = renderer
    .create(
        <VideoScreen
          filmId={FILM.id}
          time={`00:00`}
          progressValue={0}
          isPlaying={false}
          onChangePlay={handleClick}
          onClickClose={handleClick}
          onClickFullScreen={handleClick}
          onSendVideoRef={handleClick}
          onSendProgressRef={handleClick}
          film={FILM}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
