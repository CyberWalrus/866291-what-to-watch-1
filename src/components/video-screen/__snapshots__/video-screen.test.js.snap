// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoScreen correctly renders after relaunch 1`] = `
<div
  className="player"
  style={
    Object {
      "background": "url(https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg) no-repeat center center fixed",
      "backgroundSize": "cover",
      "zIndex": "100",
    }
  }
>
  <video
    className="player__video"
    controls={false}
    height="100%"
    loop={false}
    muted={false}
    poster="https://es31-server.appspot.com/wtw/static/film/background/Shutter_Island.jpg"
    src="http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4"
    width="100%"
  />
  <button
    className="player__exit"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            null,
          ],
          Array [
            null,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={0}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [
                null,
              ],
              Array [
                null,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [
                null,
              ],
              Array [
                null,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
